### ISSUE TITLE
Source file locations

### ISSUE THREAD

**Issue by kain88-de** [MDAnalysis/mdanalysis#444]
Place all source files for compiled shared object files into the same folder as the final shared object file.

`*.pyx` files and cython-generated `*.c` would be in the same directory as the `*.so`. While external dependend C/C++/Fortran libraries are in dedicate `src` and `include` folders. See the following tree as an example.

```
MDAnalysis 
      |--lib
      |   |-- _distances.so
      |   |-- distances.pyx
      |   |-- distances.c
      |-- coordinates
          |-- _dcdmodule.so
          |-- src
              |-- dcd.c
          |-- include
              |-- dcd.h
```

This is standard see [numpy](https://github.com/numpy/numpy/tree/master/numpy/linalg/lapack_lite), [scipy](https://github.com/scipy/scipy/tree/master/scipy/spatial), [scikit-learn](https://github.com/scikit-learn/scikit-learn/tree/master/sklearn/svm/src). [mdtraj](https://github.com/mdtraj/mdtraj/tree/master/mdtraj/formats/xtc)

**Comment by orbeckst** [comment]
Why not? If it makes clearer what belongs together I have no objections.

**Comment by orbeckst** [comment]
I am officially in favor of the proposal. Any other voices?

**Comment by hainm** [comment]
I always favor following common concept in ecosystem. so +1.

**Comment by kain88-de** [comment]
could we maybe add an easy tag to this?

**Comment by orbeckst** [comment]
Something like the 'repository' tag that I just added (basically, anything relating to how we organize code), or do you want something more specific?

**Comment by kain88-de** [comment]
I thought about an easy tag meaning they are bugs easy to solve. It would be a mark to show people who want to contribute what good bugs to pick are at the beginning.

**Comment by orbeckst** [comment]
Ok, I added tag [Difficulty-easy](https://github.com/MDAnalysis/mdanalysis/labels/Difficulty-easy) and tag [Difficulty-hard](https://github.com/MDAnalysis/mdanalysis/labels/Difficulty-hard) to the list; obviously, that lies somewhat in the eye of the beholder but please: tag away! 

(I purposefully omitted *Difficulty-medium* to avoid swamping issues with a default tag...)
